apps/doji-backend/src/auth/auth.service.ts

```diff
@@ -99,4 +66,57 @@ export class AuthService {
       return null
     }
   }
+
+  async sendResetPasswordEmail(email: string): Promise<void> {
+    try {
+      // Find user by email if exists
+      const user = await this.userRepo.findOneOrFail({ email: email })
+      const token = await generateRandomUserToken()
+
+      // Store reset password token in redis
+      const redisKey = generateRedisKey(
+        RedisKeyType.RESET_TOKEN,
+        serializeUserReference({ username: user.username }),
+      )
+      await this.redis.set(redisKey, token)
+      await this.redis.expire(redisKey, RESET_TOKEN_EXPIRE_DURATION_SECONDS)
+
+      // Read reset email template file
+      const resetPasswordTemplate = fs.readFileSync(
+        path.resolve(__dirname, 'assets/reset-password.hbs'),
+        'utf8',
+      )
+      const template = Handlebars.compile(resetPasswordTemplate)
+
+      // Send the email
+      const response = await this.mailgunService.createEmail(environment.mailgun.domain, {
+        from: 'Doji Support <noreply@doji.com>',
+        to: email,
+        subject: '[Doji] Reset Password Request',
+        html: template({ domain: environment.domain, token, username: user.username }),
+      })
+
+      console.log(response)
+    } catch (err) {
+      throw new NotFoundException('User not found')
+    }
+  }
+
+  async resetPassword(token: string, newPassword: string): Promise<void> {
+    const redisKey = generateRedisKey(RedisKeyType.RESET_TOKEN, token)
+    try {
+      const userRefString = await this.redis.get(redisKey)
+      const userRef = deserializeUserReference(userRefString, this.userRepo)
+      const user = await userRef.getUser()
+
+      // Set new password
+      const hashedPassword = await bcrypt.hash(newPassword, 10)
+      user.passwordHash = hashedPassword
+
+      // Save user
+      await this.userRepo.persistAndFlush(user)
+    } catch (err) {
+      throw new InvalidToken()
+    }
+  }
```

Comment:
I got 
![image](https://user-images.githubusercontent.com/16436051/159156121-984ca33d-ae1a-49bf-854c-6be21bda93f5.png)

Reply:
And I forgot to remove redis key